GeoHealthCheck/plugins/probe/wfs3.py

```diff
@@ -0,0 +1,197 @@
+from owslib.wfs import WebFeatureService
+
+from GeoHealthCheck.probe import Probe
+from GeoHealthCheck.result import Result
+
+
+class WFS3Drilldown(Probe):
+    """
+    Probe for WFS3 endpoint "drilldown": starting
+    with top endpoint: get Collections and do
+    GetItems on them etc. Using OWSLib.WebFeatureService.
+
+    TODO: needs finalization.
+    """
+
+    NAME = 'WFS3 Drilldown'
+    DESCRIPTION = 'Traverses a OGC WFS3 (REST) API endpoint by drilling down'
+    RESOURCE_TYPE = 'OGC:WFS3'
+
+    REQUEST_METHOD = 'GET'
+
+    # PARAM_DEFS = {
+    #     'drilldown_level': {
+    #         'type': 'string',
+    #         'description': 'How heavy the drilldown should be.',
+    #         'default': 'minor',
+    #         'required': True,
+    #         'range': ['minor', 'moderate', 'full']
+    #     }
+    # }
+    """Param defs"""
+
+    def __init__(self):
+        Probe.__init__(self)
+
+    def perform_request(self):
+        """
+        Perform the drilldown.
+        See https://github.com/geopython/OWSLib/blob/
+        master/tests/doctests/wfs3_GeoServerCapabilities.txt
+        """
+        wfs3 = None
+        collections = None
+
+        # 1. Test top endpoints existence
+        result = Result(True, 'Test Top Endpoints')
+        result.start()
+        try:
+            wfs3 = WebFeatureService(self._resource.url, version='3.0')
+            wfs3.conformance()
+            collections = wfs3.collections()
+        except Exception as err:
+            result.set(False, str(err))
+
+        result.stop()
+        self.result.add_result(result)
+
+        # 2. Test layers
+        # TODO: use parameters to work on less/more drilling
+        # "full" could be all layers.
+        result = Result(True, 'Test Collections')
+        result.start()
+        coll_name = ''
+        try:
+            for collection in collections:
+                coll_name = collection['name']
+                coll_name = coll_name.encode('utf-8')
+
+                try:
+                    items = wfs3.collection_items(coll_name, limit=2)
```

Comment:
Why `limit=2` and not `limit=1`, for example?

Reply:
No particular reason limit=1 would be fine as well, now that I found some WFS3 endpoints with lots of Collections.

Reply:
ok.  aside: `limit=1` is also a poor person's version of WFS1/2 `DescribeFeatureType` as well :)